### PR TITLE
overlays: init holochain-nixpkgs overlay

### PR DESCRIPTION
**Pull request by cole-h**
Supersedes https://github.com/Holo-Host/holo-nixpkgs/pull/1063.
Addresses https://github.com/Holo-Host/bugs-n-issues/issues/21.

---

Opening a new PR because GitHub is weird when you delete the base branch of a PR (I can't change the base to `develop` in https://github.com/Holo-Host/holo-nixpkgs/pull/1063...) -- sorry for the noise.

### CODE REVIEW

File: overlays/holochain-nixpkgs/default.nix [hunk 1]
```diff
@@ -3,8 +3,8 @@ let
   holochainNixpkgs = final.fetchFromGitHub {
     owner = "holochain";
     repo = "holochain-nixpkgs";
-    rev = "1e2067cecb4d7c02fc38ac407ec84f3c6e25a402";
-    sha256 = "sha256-CSEUkxMNC5VpKm75fJsYPE9Harbuy5m2xfXUOkkGqw0=";
+    rev = "8afbe20786130cf94a6f5d95293b5d6fa445cde6";
```
Comment (cole-h) [inline overlays/holochain-nixpkgs/default.nix:6]: > This is the unmerged commit from https://github.com/holochain/holochain-nixpkgs/pull/44. The git revision will probably change once it's merged.

_Originally posted by @cole-h in https://github.com/Holo-Host/holo-nixpkgs/pull/1063#discussion_r788063960_

**Review by steveeJ — APPROVED** [review]
:tada:

**Comment by cole-h** [comment]
I think this is in a good state, now!

Do note that this includes a bump of Nixpkgs in order to support the new `cargoLock.lockFile` machinery that `holochain-nixpkgs` relies upon (commit [`88180bd` (#1079)](https://github.com/Holo-Host/holo-nixpkgs/pull/1079/commits/88180bdf51620fa3f4ffa5c646c91f2674953419)).

File: overlays/holochain-nixpkgs/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  holochainMeta = final.callPackage "${holochainNixpkgs}/packages/holochain" { };
+
+  inherit (holochainMeta)
+    mkHolochainAllBinariesWithDeps
+    ;
+in
+{
+  # expose all versions to make hydra build them
+  holochainAllBinariesWithDeps = holochainMeta.holochainAllBinariesWithDeps // {
+    # In order to update this version, please take a look at the documentation
```
Comment (peeech) [inline overlays/holochain-nixpkgs/default.nix:19]: Thank you for this documentation, I was just wondering how would we maintain it in the future. Great!

**Review by peeech — APPROVED** [review]
I am merging it into develop right away as other PRs depend on it.

**Comment by peeech** [comment]
Lovely, nightly tests are passing!

**Pull request merged**